src/Engine/Renderer/Renderer.hpp

```diff
@@ -387,8 +387,8 @@ class RA_ENGINE_API Renderer
     uint m_width {0};
     uint m_height {0};
 
-    ShaderProgramManager* m_shaderMgr {nullptr};
-    RenderObjectManager* m_roMgr {nullptr};
+    ShaderProgramManager* m_shaderProgramManager {nullptr};
```

Comment:
In addition to comment in .cpp, we can comment here to clearly state something like 
```suggestion
/// This ptr is own and initialized by the engine, and remains valid during all program execution.
/// It is set in initialize, called after engine initialization.
    ShaderProgramManager* m_shaderProgramManager {nullptr};
```